packages/library-sync/src/commonMain/kotlin/io/realm/kotlin/mongodb/sync/SyncConfiguration.kt

```diff
@@ -283,6 +312,24 @@ public interface SyncConfiguration : Configuration {
         public fun errorHandler(errorHandler: SyncSession.ErrorHandler): Builder =
             apply { this.errorHandler = errorHandler }
 
+        /**
+         * Sets the strategy that would handle the client reset by this synced Realm. In the case that
```

Comment:
We should also document the restrictions on types here as this is the main configuration point.